doc/sphinx/source/inspection/debug_core.rst

```diff
@@ -0,0 +1,105 @@
+.. inspection/debug_core.rst:
+
+.. contents::
+
+.. _debug_core:
+
+Debug & Build Options
+#####################
+
+Build nGraph with various compile flags and environment variables to debug your 
+use case.
+
+
+Compile Flags
+=============
+
+.. csv-table::
+   :header: "Compile Flag", "Description", "Default Value"
+   :widths: 20, 35, 5
+   :escape: ~
+
+   ``NGRAPH_UNIT_TEST_ENABLE``, Control the building of unit tests,``TRUE``
+   ``NGRAPH_TOOLS_ENABLE``, Control the building of tools,``TRUE``
+   ``NGRAPH_CPU_ENABLE``, Control the building of the CPU backend,``TRUE``
+   ``NGRAPH_INTERPRETER_ENABLE``, Control the building of the ``INTERPRETER`` backend,``TRUE``
+   ``NGRAPH_NOP_ENABLE``, Control the building of the NOP backend,``TRUE``
+   ``NGRAPH_ENABLE_CPU_CONV_AUTO``, Enable mkldnn convolution_auto for CPU,``TRUE``
+   ``NGRAPH_JSON_ENABLE``, Enable JSON based serialization and tracing features,``TRUE``
+   ``NGRAPH_DYNAMIC_COMPONENTS_ENABLE``, Enable dynamic loading of components,``TRUE``
+   ``NGRAPH_FAST_MATH_ENABLE``, Enable fast math, ``ON``
+   ``NGRAPH_TBB_ENABLE``, "Only if (``NGRAPH_CPU_ENABLE``) Control usage of TBB for CPU backend",``TRUE``
+   ``NGRAPH_USE_PREBUILT_LLVM``, ,``FALSE``
+   ``NGRAPH_ONNX_IMPORT_ENABLE``, Enable ONNX importer,``FALSE``
+   ``NGRAPH_DEBUG_ENABLE``, Enable output for ``NGRAPH_DEBUG`` statements,``FALSE``
+   ``NGRAPH_DEX_ONLY``, Build CPU DEX without codegen,``FALSE``
+   ``NGRAPH_MLIR_ENABLE``, Control the building of MLIR backend,``FALSE``
+   ``NGRAPH_INTELGPU_ENABLE``, Control the building of the Intel GPU backend with clDNN,``FALSE``
+   ``NGRAPH_GPU_ENABLE``, Control the building of the GPU backend,``FALSE``
+   ``NGRAPH_GPUH_ENABLE``, Control the building of the Hybrid GPU backend,``FALSE``
+   ``NGRAPH_GENERIC_CPU_ENABLE``, Enable build ``NGRAPH`` for generic CPU backend,``FALSE``
+   ``NGRAPH_DEPRECATED_ENABLE``, Enable compiler deprecation pragmas for deprecated APIs (recommended only for development use),``FALSE``
+   ``NGRAPH_CODE_COVERAGE_ENABLE``, Enable code coverage data collection,``FALSE``
+   ``NGRAPH_LIB_VERSIONING_ENABLE``, Enable shared library versioning,``FALSE``
+   ``NGRAPH_PYTHON_BUILD_ENABLE``, Enable build of ``NGRAPH`` python package wheel,``FALSE``
+   ``NGRAPH_PLAIDML_ENABLE``, Enable the PlaidML backend, ``${PLAIDML_FOUND}``
+   ``NGRAPH_STATIC_LIB_ENABLE``, Enable build ``NGRAPH`` static library,``FALSE``
+   ``NGRAPH_INTERPRETER_STATIC_LIB_ENABLE``, Enable build INTERPRETER backend static library,``FALSE``
+   ``NGRAPH_CPU_STATIC_LIB_ENABLE``, Enable build CPU backend static library,``FALSE``
+   ``NGRAPH_DISTRIBUTED_ENABLE``, Enable distributed training using MLSL/OpenMPI,``OFF``
+   ``NGRAPH_DISTRIBUTED_MLSL_ENABLE``, Use MLSL ,``OFF``
+   ``NGRAPH_HALIDE``, , ``OFF``
+   ``NGRAPH_DOC_BUILD_ENABLE``, Automatically build documentation ,``OFF``
+
+
+Environment Variables
+=====================
+
+.. csv-table::
```

Comment:
can you please add a note here that "these flags are subject to change"? else LGTM